packages/legend-query/src/stores/QueryBuilderMilestoningHelper.ts

```diff
@@ -0,0 +1,447 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  getNullableFirstElement,
+  guaranteeNonNullable,
+  guaranteeType,
+  UnsupportedOperationError,
+} from '@finos/legend-shared';
+import {
+  Class,
+  type PureModel,
+  type ValueSpecification,
+  DerivedProperty,
+  getMilestoneTemporalStereotype,
+  MILESTONING_STEREOTYPE,
+  DEFAULT_PROCESSING_DATE_MILESTONING_PARAMETER_NAME,
+  AbstractPropertyExpression,
+  Association,
+  TYPICAL_MULTIPLICITY_TYPE,
+  DEFAULT_BUSINESS_DATE_MILESTONING_PARAMETER_NAME,
+  INTERNAL__PropagatedValue,
+  PrimitiveInstanceValue,
+  PRIMITIVE_TYPE,
+  VariableExpression,
+} from '@finos/legend-graph';
+import type { QueryBuilderDerivedPropertyExpressionState } from './QueryBuilderPropertyEditorState';
+import type { QueryBuilderSetupState } from './QueryBuilderSetupState';
+import {
+  functionExpression_setParametersValues,
+  propertyExpression_setParametersValue,
+} from './QueryBuilderValueSpecificationModifierHelper';
+import type { QueryBuilderState } from './QueryBuilderState';
+import { QueryBuilderValueSpecificionCloneHelper } from './QueryBuilderValueSpecificationCloneHelper';
+
+export const valueSpecifiation_isEqual = (
+  param1: ValueSpecification | undefined,
```

Comment:
Let's reason about the usefulness of this method in the final round of review. For now it can be left here.

Reply:
We don't need to `export` this

Reply:
I think it's safe to inline this method in `checkForMilestoningParameterEquality` and rename it to `checkIfEquivalent`